src/Symfony/Component/Serializer/Normalizer/PropertyNormalizer.php

```diff
@@ -150,4 +152,27 @@ protected function setAttributeValue($object, $attribute, $value, $format = null
 
         $reflectionProperty->setValue($object, $value);
     }
+
+    /**
+     * @param string|object $classOrObject
+     * @param string        $attribute
+     *
+     * @return \ReflectionProperty
+     *
+     * @throws \ReflectionException
+     */
+    private function getReflectionProperty($classOrObject, $attribute)
+    {
+        $class = is_string($classOrObject) ? $classOrObject : get_class($classOrObject);
+        while (true) {
+            try {
+                return isset($reflectionClass) ? $reflectionClass->getProperty($attribute) : new \ReflectionProperty($class, $attribute);
+            } catch (\ReflectionException $e) {
+                $reflectionClass = new \ReflectionClass($class);
```

Comment:
Looks wrong to me as you are always overriding `$reflectionClass` with `$class`. This statement should only be executed if `$reflectionClass` is not set. I would add a test with a more complex hierarchy: class < parent < parent. Current code would loop forever.